ratik/src/main/java/org/opencadc/ratik/InventoryValidator.java

```diff
@@ -0,0 +1,157 @@
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.ratik;
+
+import ca.nrc.cadc.db.DBUtil;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.reg.Capabilities;
+import ca.nrc.cadc.reg.Capability;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import javax.sql.DataSource;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.inventory.db.version.InitDatabase;
+import org.opencadc.inventory.util.ArtifactSelector;
+import org.opencadc.inventory.util.BucketSelector;
+
+/**
+ * Validate local inventory.
+ */
+public class InventoryValidator implements Runnable {
+    private static final Logger log = Logger.getLogger(InventoryValidator.class);
+
+    private final ArtifactDAO artifactDAO;
+    private final URI resourceID;
+    private final ArtifactSelector artifactSelector;
+    private final BucketSelector bucketSelector;
+    private final boolean trackSiteLocations;
+
+    /**
+     * Constructor.
+     *
+     * @param daoConfig          config map to pass to cadc-inventory-db DAO classes
+     * @param resourceID         identifier for the remote query service
+     * @param artifactSelector   artifact selector implementation
+     * @param bucketSelector     uri buckets
+     * @param trackSiteLocations local site type
+     */
+    public InventoryValidator(Map<String, Object> daoConfig, URI resourceID, ArtifactSelector artifactSelector,
+                              BucketSelector bucketSelector, boolean trackSiteLocations) {
+        InventoryUtil.assertNotNull(InventoryValidator.class, "daoConfig", daoConfig);
+        InventoryUtil.assertNotNull(InventoryValidator.class, "resourceID", resourceID);
+        InventoryUtil.assertNotNull(InventoryValidator.class, "artifactSelector", artifactSelector);
+        InventoryUtil.assertNotNull(InventoryValidator.class, "bucketSelector", bucketSelector);
+        InventoryUtil.assertNotNull(InventoryValidator.class, "trackSiteLocations", trackSiteLocations);
```

Comment:
primitive `boolean` can't be null so the check on trackSiteLocations is not needed

Reply:
Removed the check.